src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -416,9 +417,10 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 		}
 	}
 
-	if (revents == G_IO_IN) {
-		if (serial_read_nonblocking(serial, &byte, 1) != 1)
-			return FALSE;
+	received_a_byte = FALSE;
+	while (revents == G_IO_IN &&
+	       serial_read_nonblocking(serial, &byte, 1) == 1) {
```

Comment:
As mentioned before, I have a OLS clone with 1MB of memory. This may be a bit too much to poll in one go? Dunno what a reasonable size for chunks would be? 64KB?

Reply:
I agree to not block the event loop, though the device would have respond pretty quickly to not cause `serial_read_nonblocking` to return nothing (assuming the latter really doesn't block). Nonetheless, to be safe, I added a safeguard to "ols: Capture multiple bytes at once".